### PR TITLE
[kube-state-metrics] Use digest instead of sha

### PR DESCRIPTION
**Pull request by sathieu**
#### What this PR does / why we need it

sha is for digest without digest type.

digest is more commonly used, and prepare for possible future digest types

Same as #3130.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-state-metrics/Chart.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ keywords:
 - prometheus
 - kubernetes
 type: application
-version: 5.5.0
+version: 5.5.1
```
[the remark this answers is not in the document]
Comment (sathieu) [reply charts/kube-state-metrics/Chart.yaml]: @mrueg I've bumped major and added notes in `README.md`. Back to you 🏸 !

File: charts/kube-state-metrics/README.md [hunk 2]
```diff
@@ -40,12 +40,9 @@ helm upgrade [RELEASE_NAME] prometheus-community/kube-state-metrics [flags]
...
 _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
 
-### Migrating from stable/kube-state-metrics and kubernetes/kube-state-metrics
-
-You can upgrade in-place:
+## From 5.x to 6.x
 
-1. [get repository info](#get-repository-info)
-1. [upgrade](#upgrading-chart) your existing release name using the new chart repository
+This version renames `image.sha` to `image.digest` and `kubeRBACProxy.image.sha` to `kubeRBACProxy.image.sha`.
```
Comment (mrueg) [inline charts/kube-state-metrics/README.md:45]:
```suggestion
This version renames `image.sha` to `image.digest` and `kubeRBACProxy.image.sha` to `kubeRBACProxy.image.digest`.
```

Comment (sathieu) [reply charts/kube-state-metrics/README.md]: fixed. Back to you @mrueg 🏓 !

**Comment by dotdc** [comment]
@sathieu There's a conflict on `charts/kube-state-metrics/Chart.yaml`.
Could you rebase your branch on `main`?

**Comment by sathieu** [comment]
@dotdc Done. Sorry for the noise.

(I've thought I had done it, but forgot to rebase probably)

**Comment by dotdc** [comment]
@sathieu Just opened https://github.com/prometheus-community/helm-charts/issues/3355, feel free to jump in the discussion.

**Comment by dotdc** [comment]
According to https://github.com/prometheus-community/helm-charts/issues/3355, we can make the choice for kube-state-metrics.
I'm fine with both honestly.

@mrueg Do you think we should merge this?

**Comment by mrueg** [comment]
@dotdc naming is hard and to be honest, I don't know if it really matters to be more correct here. There is no "helm fmt" to enforce certain standards (although this probably would be a good idea).
I agree though with https://github.com/prometheus-community/helm-charts/issues/3355#issuecomment-1541553294 it should be unified to avoid folks guessing.

**Comment by dotdc** [comment]
After reading the comments, I don't think it's the good time do it.
Maybe we'll include this next to another breaking change later.
Closing

**Pull request closed without merging**